### PR TITLE
Fixes AI radio and a bunch of backend changes

### PR DESCRIPTION
**Pull request by MaxIsJoe**
Alright, this PR turned out to be much bigger than anticipated just for this simple fix, so I've decided to leave the rest of my plans for later.
This PR introduces a bunch of backend changes for the telecomms rework and signals. Also finally starts working on implementing modules to public terminals.

All of these changes work together to implement physical radios for the AI which is embedded in a public terminal.

### Changelog:
CL: [New] Signal Emitters can now send multiple types of signals.
CL: [New] Began work on public terminal modules, nothing is intractable yet except for the AI.
CL: [Fix] All non-standard humanoid players who do not possess a dynamic inventory will no longer receive an NRE for being built different.
CL: [Fix] Fixed an old, annoying NRE involving the AI trying to access HealthV2 when it didn't have it.
CL: [Fix] The AI now can send signals just like the rest of the crew to be able to use radios via a public terminal added to it's core chamber.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Chat/Chat.cs [hunk 1]
```diff
@@ -60,11 +65,28 @@ public static void InvokeChatEvent(ChatEvent chatEvent)
 				chatEvent.channels = channel;
 				if (chatEvent.originator.TryGetComponent<PlayerScript>(out var playerScript))
 				{
+					//There are some cases where the player might not have a dynamic item storage (like the AI)
+					if (playerScript.DynamicItemStorage == null)
+					{
+						Physics2D.OverlapCircleNonAlloc(playerScript.mind.body.AssumedWorldPos.To2(), searchRadiusForSphereResult, nonAllocPhysicsSphereResult);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Core/Chat/Chat.cs]: The Ai's body is the camera not the core so this wont work correctly as it'll send to everything around the camera not the core.

playerScript.PlayerChatLocation.AssumedWorldPos();

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.cs]: good catch, working on it as well as the test fail soon.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/Chat.cs]: Done

**Comment by corp-0** [comment]
make your public fields private as suggested by codacy and it is done

**Comment by ThatDan123** [comment]
Did you also forget Outpost Station, i think that has AI too?

**Comment by MaxIsJoe** [comment]
is outpost in main rotation?

**Comment by ThatDan123** [comment]
I think it is still

**Comment by MaxIsJoe** [comment]
alright i fixed all issues i found, merge pls

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**